microsoft/testsuites/vm_extensions/cvt.py

```diff
@@ -0,0 +1,274 @@
+        )
+
+        file_list += find_tool.find_files(
+            test_dir,
+            name_pattern="*.txt",
+            file_type="f",
+            sudo=True,
+            ignore_not_exist=True,
+        )
+
+        for file in file_list:
+            log.info(f"Copying file {file} to {log_path}")
+            try:
+                file_name = file.split("/")[-1]
+                node.shell.copy_back(
+                    node.get_pure_path(file),
+                    log_path / f"{file_name}",
+                )
+            except FileNotFoundError:
+                log.error(f"File {file} doesn't exist.")
+
+    def run_cvt_tests(
+        self,
+        log: Logger,
+        node: Node,
+        log_path: Path,
+        variables: Dict[str, Any],
+    ) -> Optional[int]:
+        cvt_bin = "indskflt_ct"
+        sas_uri = variables.get("cvt_binary_sas_uri", "")
+        if not sas_uri:
+            log.error("sas uri for cvt binary is empty.")
+            return 1
+        cvt_root_dir = str(node.working_path) + "/LisaTest/"
+        cvt_download_dir = cvt_root_dir + "cvt_files/"
+
+        mkdir = node.tools[Mkdir]
+        wget = node.tools[Wget]
+
+        mkdir.create_directory(cvt_download_dir, sudo=True)
+        download_path = wget.get(
+            url=f"{sas_uri}",
+            filename=cvt_bin,
+            file_path=cvt_download_dir,
+            sudo=True,
+        )
+
+        cvt_md5sum = node.execute(
+            f"md5sum {download_path}", shell=True, sudo=True
+        )
+        log.info(f"md5sum '{download_path}' : '{cvt_md5sum}'")
+
+        result = self.run_script(node=node, log=log, test_dir=cvt_download_dir)
+        self.copy_cvt_logs(
+            node=node,
+            log=log,
+            test_dir=Path(node.working_path.parent.parent),
+            log_path=log_path,
+        )
+        return result.exit_code
+
+    @TestCaseMetadata(
+        description="""
+        this test validate the functionality of ASR driver by verifying
+        integrity of a source disk with respect to a target disk
+        """,
+        priority=1,
+        use_new_environment=True,
+        timeout=TIMEOUT,
+    )
+    def run_cvt(
+        self,
+        node: Node,
+        log: Logger,
+        log_path: Path,
+        variables: Dict[str, Any],
+    ) -> None:
+        info = node.tools[Uname].get_linux_information()
+        log.info(
+            f"release: '{info.uname_version}', "
+            f"version: '{info.kernel_version_raw}', "
+            f"hardware: '{info.hardware_platform}', "
+            f"os: '{info.operating_system}'"
+        )
+
+        self.init_disk(node=node, log=log)
+        os = self.get_os_info_from_extension(node=node, log=log)
+        self.install_asr_extension_distro(node=node, log=log, os=os)
+        result = self.run_cvt_tests(
+            node=node, log=log, log_path=log_path, variables=variables
+        )
+        assert_that(result).is_equal_to(0)
```

Comment:
Add `described_as` to explain the error. And it's better to extra the error messages to the assertion, so detail failure message is easier to discover.